hawc/apps/animalv2/models.py

```diff
@@ -0,0 +1,576 @@
+
+    # TODO - flat_complete_header_row / flat_complete_data_row
+
+
+"""
+based on conversations with Vidhi and Pam - various fields here (n, treatment_name, loael, etc.)
+are duplicated on both the parent dataExtraction object and these associated group/animal level data
+objects. Not sure how this is supposed to work - does the dataExtraction level field take precedence?
+Provide a default that can be overridden? Still chatting with them to see what's expected here so
+that type of overlap is the most likely portion of the model to be subject to change.
+"""
+
+
+class DoseResponseGroupLevelData(models.Model):
+    objects = managers.DoseResponseGroupLevelDataManager()
+
+    data_extraction = models.ForeignKey(
+        DataExtraction, on_delete=models.CASCADE, related_name="v2_group_level_data"
+    )
+    treatment = models.ForeignKey(
+        Treatment, on_delete=models.CASCADE, related_name="v2_group_level_data"
+    )
+    # specifically, DoseGroup.dose?
+    dose = models.ForeignKey(
+        DoseGroup, on_delete=models.CASCADE, related_name="v2_group_level_data"
+    )
+    n = models.PositiveSmallIntegerField(blank=True, null=True, validators=[MinValueValidator(0)])
+    response = models.FloatField()
+    variance = models.FloatField(blank=True, null=True, validators=[MinValueValidator(0)])
+    treatment_related_effect = models.PositiveSmallIntegerField(
+        choices=constants.TreatmentRelatedEffect.choices
+    )
+    statistically_significant = models.PositiveSmallIntegerField(
+        choices=constants.StatisticallySignificant.choices
+    )
+    p_value = models.CharField(max_length=128, blank=True, help_text="TODO")
+    NOAEL = models.SmallIntegerField(default=-999, help_text="No observed effect level")
+    LOAEL = models.SmallIntegerField(default=-999, help_text="Lowest observed effect level")
+
+    created = models.DateTimeField(auto_now_add=True)
+    last_updated = models.DateTimeField(auto_now=True)
+
+    def get_assessment(self):
+        return self.data_extraction.get_assessment()
+
+    def get_study(self):
+        return self.data_extraction.get_study()
+
+    # TODO - flat_complete_header_row / flat_complete_data_row
+
+
+class DoseResponseAnimalLevelData(models.Model):
+    objects = managers.DoseResponseAnimalLevelDataManager()
+
+    data_extraction = models.ForeignKey(
+        DataExtraction, on_delete=models.CASCADE, related_name="v2_animal_level_data"
+    )
+    cage_id = models.CharField(max_length=128, blank=True, help_text="TODO")
+    animal_id = models.CharField(max_length=128, blank=True, help_text="TODO")
+    response = models.FloatField()
+
+    created = models.DateTimeField(auto_now_add=True)
+    last_updated = models.DateTimeField(auto_now=True)
+
+    def get_assessment(self):
+        return self.data_extraction.get_assessment()
+
+    def get_study(self):
+        return self.data_extraction.get_study()
+
+    # TODO - flat_complete_header_row / flat_complete_data_row
```

Comment:
we can remove these TODO from all models; we're no longer implementing using these methods